File: app/src/main/java/com/example/ep2024/domain/repository/AuthRepository.java
package com.example.ep2024.domain.repository;

import com.example.ep2024.data.model.requestDTOs.auth.LoginRequestDTO;
import com.example.ep2024.data.model.responseDTOs.auth.LoginResponseDTO;

public interface AuthRepository {

    LoginResponseDTO login(LoginRequestDTO req) throws Exception;

}

File: app/src/main/java/com/example/ep2024/domain/repository/AuthRepository.kt
package com.example.ep2024.domain.repository

import com.example.ep2024.data.model.requestDTOs.auth.LoginRequestDTO
import com.example.ep2024.data.model.requestDTOs.auth.RegisterRequestDTO
import com.example.ep2024.data.model.responseDTOs.auth.CommonMessageResponseDTO
import com.example.ep2024.data.model.responseDTOs.auth.LoginResponseDTO

interface AuthRepository {

    @Throws(Exception::class)
    fun login(req: LoginRequestDTO?): LoginResponseDTO?

    @Throws(Exception::class)
    fun signup(req: RegisterRequestDTO?): CommonMessageResponseDTO?

}
